runtime/vm/FFITypeHelpers.hpp

```diff
@@ -112,17 +112,20 @@ class FFITypeHelpers
 
 		*typeFFI = NULL;
 		UDATA structSize = UDATA_MAX;
-		char bufOnStackLocal[J9VM_LAYOUT_STRING_ON_STACK_LIMIT];
-		char *layout = copyStringToUTF8WithMemAlloc(_currentThread, layoutStringObject, J9_STR_NULL_TERMINATE_RESULT, "", 0, bufOnStackLocal, J9VM_LAYOUT_STRING_ON_STACK_LIMIT, NULL);
+		char layoutBuffer[J9VM_LAYOUT_STRING_ON_STACK_LIMIT];
+		char* layout = copyStringToUTF8WithMemAlloc(_currentThread, layoutStringObject, J9_STR_NULL_TERMINATE_RESULT, "", 0, layoutBuffer, sizeof(layoutBuffer), NULL);
+
+		/* Subsequent calls will modify the contents of this pointer, so preserve the original pointer for j9mem_free_memory */
+		char* layoutTemp = layout;
```

Comment:
Technically this could be moved after the null check.

Reply:
>Technically this could be moved after the null check.
@pshipton this is not possible because of the `goto`, otherwise we get the following error:

>FFITypeHelpers.hpp: In member function 'UDATA FFITypeHelpers::getCustomFFIType(ffi_type**, j9object_t)':
>FFITypeHelpers.hpp:127:1: error: jump to label 'doneGetCustomFFIType' [-fpermissive]
>FFITypeHelpers.hpp:119:9: error:   from here [-fpermissive]
>FFITypeHelpers.hpp:123:9: error:   crosses initialization of 'char* layoutTemp'

Reply:
ok